### PR TITLE
DCS-1702 add notifyClient and notificationService

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/routes/feedback/feedbackValidation.test.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  it('Valid input', () => expect(FeedbackValidator({ feedback: 'Some content', email: 'a.user@email' })).toEqual([]))
+
+  test.each([
+    ['Abc.example.com'],
+    ['A@b@c@example.com'],
+    ['a"b(c)d,e:f;g<h>i[jk]l@example.com'],
+    ['just"not"right@example.com'],
+    // eslint-disable-next-line prettier/prettier, no-useless-escape
+    ['this is"notallowed@example.com '],
+    // eslint-disable-next-line prettier/prettier, no-useless-escape
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/feedback/feedbackValidation.test.ts:29]: Ah I should have left a comment on these - it's the only way the linter would let me test that the back slash would be caught by the validation checks in a user input email address.

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/feedback/feedbackValidation.test.ts:29]: Ah the linter as part of the commit command removed the \s automatically. Good to know the validation checks for this but I'll remove the disable comments as they are not actually needed if we can't commit a \

File: server/services/notificationService.test.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import NotificationService, { type EmailPersonalisation } from './notificationService'
+import config from '../config'
+
+const notifyApi = {
```
Comment (connormaglynn) [inline server/services/notificationService.test.ts:4]: Might be better to stick with notifyClient naming convention? 🤔

File: server/services/notificationService.test.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  })
+
+  describe('Send feedback email', () => {
+    const feedbackEmail: EmailPersonalisation = {
+      username: 'A_USER',
+      prison: 'MDI',
+      feedback: 'Some feedback',
+    }
+
+    it('Details are retrieved for user', async () => {
```
Comment (connormaglynn) [inline server/services/notificationService.test.ts:26]: Think we can remove this test? Or add some assertions? For example, that the email is sent when optional fields (email) aren't passed? 😃

File: server/routes/feedback/feedbackValidation.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/feedback/feedbackValidation.test.ts]: Comments now removed

[the remark this answers is not in the document]
Comment (connormaglynn) [reply server/routes/feedback/feedbackValidation.test.ts]: Ah I see what you mean. Might have to escape the backslashes so they're passed through within the string i.e.`'this\\ still\\"not\\\\allowed@example.com'` comes through into the test as `this\ still\"not\\allowed@example.com`, which is what get's validated. If that's what we're after? 👀

File: server/services/notificationService.test.ts (hunk 2 above)
Comment (louisenorris) [reply server/services/notificationService.test.ts]: Amended as suggested

File: server/services/notificationService.test.ts (hunk 3 above)
Comment (louisenorris) [reply server/services/notificationService.test.ts]: Ah, realised in the service layer the email isn't actually optional as a default value is assigned and passed down from the controller is no email address is provided by user - have made amendments for this in the lates commit and removed this no unnecessary test

**Review by connormaglynn — APPROVED** [review]
Some great work here @louisenorris! 🚀  just a few minor comments, but overall lgtm 👍

File: server/routes/feedback/feedbackValidation.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/routes/feedback/feedbackValidation.test.ts]: Escaping the backslashes doesn't seem to have that result, might just leave this bit as it is now unless you object @connormaglynn ?

**Pull request merged**